### PR TITLE
Visual Studio 2022: Support building liblouis on Clang 15 and above

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
https://github.com/liblouis/liblouis/issues/1259

### Summary of the issue:
Starting with Clang 15, liblouis no longer compiles because Clang 15 treats int to pointer conversion as erroneous.

### Description of user facing changes
NVDA builds properly again on VS 2022 preview.

### Description of development approach
Added a flag that degrades the error to a warning again.

### Testing strategy:
Ensure that NVDA builds correctly.

### Known issues with pull request:
This is not strictly necessary at this moment because Clang 15 hasn't yet landed in stable. Please note however that we have no control over the appveyor release process and that building will instantly break without this patch as soon as Appveyor updates VS to a version that has Clang 15.

### Change log entries:
none needed

### Code Review Checklist:

- [x] Pull Request description:
  - description is up to date
  - change log entries
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] API is compatible with existing add-ons.
- [x] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] Security precautions taken.

### CODE REVIEW

**Comment by michaelDCurran** [comment]
So I take it that as because this pr built okay on VS 2019, and we can confirm VS 2022 stable once pr #14313  is merged, then we may as well merge this also once we are on VS 2022.

**Comment by leonardder** [comment]
Yes, correct. I would personally do that as that ensures we don't have any outage as osoon as Appveyor updates their image.

**Review by seanbudd — APPROVED** [review]

File: nvdaHelper/liblouis/sconscript
Comment (seanbudd) [inline nvdaHelper/liblouis/sconscript:63]: should this be removed? why?

Comment (leonardder) [reply nvdaHelper/liblouis/sconscript:63]: Ah I'm sorry, I motivated this slightly https://github.com/nvaccess/nvda/pull/14313#issuecomment-1297028520 but forgot to mention it here.
The short story is that we are not treating warnings as errors for liblouis, so there is no point in changing the warning level, especially since with and without the warning level replacement, the number of warnings is equal anyway.

**Comment by leonardder** [comment]
Visual Studio 2022 17.4 was release today. If I'm not mistaken, this is the first version containing Clang 15 and therefore builds of master will fail after updating.

**Comment by michaelDCurran** [comment]
to clarify, all builds of NVDA will fail locally. Note that appveyor is 
still on VS 2019 until we can resolve differences with Notepad on 
Windows 10 / 11 in our system tests.

**Pull request merged**
